skipruntime-ts/ts/src/skipruntime_api.ts

```diff
@@ -581,3 +402,76 @@ export interface SKStore extends Constant {
 
   log(object: TJSON): void;
 }
+
+export type Token = {
+  duration: number;
+  value: string;
+};
+
+export type Entry<K extends TJSON, V extends TJSON> = [K, V[]];
+
+export type ReactiveResponse = {
+  collection: string;
+  watermark: string;
+};
+
+export type Watermaked<K extends TJSON, V extends TJSON> = {
+  values: Entry<K, V>[];
+  watermark: string;
+  update?: boolean;
+};
+
+export type Notifier<K extends TJSON, V extends TJSON> = (
+  values: Entry<K, V>[],
+  watermark: string,
+  update: boolean,
+) => void;
+
+export type SkipBuilder = (
+  iCollection: Record<string, CollectionWriter<TJSON, TJSON>>,
+) => SkipRuntime;
+
+export interface CollectionWriter<K extends TJSON, V extends TJSON> {
+  write(key: K, value: V[]): void;
+  writeAll(values: Entry<K, V>[]): void;
+  delete(keys: K[]): void;
+}
+
+export interface SkipRuntime {
+  // READ
+  getAll<K extends TJSON, V extends TJSON>(
+    resource: string,
+    params: JSONObject,
+    reactiveAuth?: Uint8Array | string,
+  ): { values: Entry<K, V>[]; reactive?: ReactiveResponse };
+  head(
+    resource: string,
+    params: JSONObject,
+    reactiveAuth: Uint8Array | string,
+  ): ReactiveResponse;
+  getOne<V extends TJSON>(
+    resource: string,
+    params: JSONObject,
+    key: string | number,
+  ): V[];
+  // WRITE
+  put<V extends TJSON>(
+    collection: string,
+    key: string | number,
+    value: V[],
+  ): void;
+
+  patch<K extends TJSON, V extends TJSON>(
+    collection: string,
+    values: Entry<K, V>[],
+  ): void;
+
+  delete(collection: string, key: string | number): void;
```

Comment:
`delete` is a keyword in JS -- I think we should avoid it, seems like this might cause some issues for people, at least weird highlighting but potentially unintended semantics.